base/linalg/dense.jl

```diff
@@ -358,6 +358,10 @@ kron(a::AbstractVector, b::AbstractMatrix) = kron(reshape(a, length(a), 1), b)
 
 # Matrix power
 (^)(A::AbstractMatrix, p::Integer) = p < 0 ? Base.power_by_squaring(inv(A), -p) : Base.power_by_squaring(A, p)
+function (^)(A::AbstractMatrix{T}, p::Integer) where T<:Integer
+    TT = Base.promote_op(^, T, typeof(p))
```

Comment:
Why is it necessary to promote here? I don't think this version is required. Couldn't you just change the previous method to
```
(^)(A::AbstractMatrix, p::Integer) = Base.power_by_squaring(A, p)
```
?